### PR TITLE
FE: Dashboard Skeleton

### LINKED ISSUE #8: FE: Dashboard Skeleton
- Adapt the current project skeleton to match project mockups
- The project skeleton will have the general frontend structure of the project, but without the specific components filled out
- If there is a navbar in the mockups, the skeleton should include the navbar + content pages for the app for a logged in user experience
- The navbar should be persistent across the whole logged in user experience and include a placeholder image for the user
- Dropdown for profile in navbar should show two links:
    - logout (already implemented on the backend)
    - go to profile

### PR DESCRIPTION
**Pull request by samimsu**
### What this PR does (required):
- creates the dashboard skeleton
- comments out auth code bits to allow dashboard viewing without being logged in

### Screenshots / Videos (required):
image
image

### Any information needed to test this feature (required):
- `cd client`
- `npm start`
- Navigate to http://localhost:3000/dashboard

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ function App(): JSX.Element {
     <MuiThemeProvider theme={theme}>
       <BrowserRouter>
         <SnackBarProvider>
-          <AuthProvider>
+          {/*<AuthProvider>*/}
```
Comment (bonnieli) [inline client/src/App.tsx]: any reason to comment AuthProvider?

File: client/src/components/DashboardHeader/DashboardHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import CalendarTodayOutlinedIcon from '@material-ui/icons/CalendarTodayOutlined';
+import AddIcon from '@material-ui/icons/Add';
+import Avatar from '@material-ui/core/Avatar';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import React from 'react';
+import logo from '../../Images/logo.png'
+
+const DashboardHeader = (): JSX.Element => {
+    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
```
Comment (bonnieli) [inline client/src/components/DashboardHeader/DashboardHeader.tsx]: looks like some linting issues - I suggest syncing with the team to make sure 
1. Everyone has linters installed (eslint & prettier)
2. Everyone has the right settings :)

File: client/src/pages/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -1,38 +1,51 @@
...
+// import AppBar from '@material-ui/core/AppBar';
+// import Toolbar from '@material-ui/core/Toolbar';
+// import CircularProgress from '@material-ui/core/CircularProgress';
 import useStyles from './useStyles';
-import { useAuth } from '../../context/useAuthContext';
-import { useSocket } from '../../context/useSocketContext';
-import { useHistory } from 'react-router-dom';
-import ChatSideBanner from '../../components/ChatSideBanner/ChatSideBanner';
-import { useEffect } from 'react';
+// import { useAuth } from '../../context/useAuthContext';
```
Comment (bonnieli) [inline client/src/pages/Dashboard/Dashboard.tsx]: bunch of commented out code here

**Review by bonnieli — COMMENTED** [review]
On a design perspective, let's try to match it a bit more like the mocks

Some suggestions
- More padding in the nav bar
- The hamburger icon is on the right hand side 
- Dashboard and Calendar link is more centered
- Create board and display pic is more to the right
- Create board button is blue

File: client/src/App.tsx (hunk 1 above)
Comment (samimsu) [reply client/src/App.tsx]: So I can access http://localhost:3000/dashboard without being rerouted to the login page.

File: client/src/components/DashboardHeader/DashboardHeader.tsx (hunk 2 above)
Comment (samimsu) [reply client/src/components/DashboardHeader/DashboardHeader.tsx]: I have linters installed too. I was having issues in the beginning so I changed my .eslintrc. I'm not sure which issues you're referencing. I didn't push my .eslintrc file. Maybe I should do that.

File: client/src/pages/Dashboard/Dashboard.tsx (hunk 3 above)
Comment (samimsu) [reply client/src/pages/Dashboard/Dashboard.tsx]: Yeah just so I could access the dashboard without worrying about auth.

Comment (sina-jamshidi) [reply client/src/pages/Dashboard/Dashboard.tsx]: When pushing code to main we want to avoid commented out code. If it needs to be commented it can be removed :) We can always add it back in later if we need to.
In this case I would say you should uncomment `useAuth`. Remember that main is where all your teammates will also be pulling from, so we want to make sure it is the most up to date running version of our app!

File: client/src/App.tsx (hunk 1 above)
Comment (sina-jamshidi) [reply client/src/App.tsx]: ^Same as my comment below

File: client/src/pages/Dashboard/Dashboard.tsx (hunk 3 above)
Comment (samimsu) [reply client/src/pages/Dashboard/Dashboard.tsx]: That makes sense. I'll fix it.

**Comment by samimsu** [comment]
> On a design perspective, let's try to match it a bit more like the mocks
> 
> Some suggestions
> 
> * More padding in the nav bar
> * The hamburger icon is on the right hand side
> * Dashboard and Calendar link is more centered
> * Create board and display pic is more to the right
> * Create board button is blue

image
image
image

File: client/src/components/DashboardAppBar/useStyles.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { makeStyles } from '@material-ui/core/styles';
+
+const primaryColor = '#759CFC';
```
Comment (bonnieli) [inline client/src/components/DashboardAppBar/useStyles.ts]: we should be putting this in the theme rather than just a constant so it can be used widely acrosss the app

**Review by bonnieli — COMMENTED** [review]
just one more comment on the theme :)

**Review by bonnieli — APPROVED** [review]

File: client/package.json [hunk 5]
```diff
@@ -31,7 +31,7 @@
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
-    "lint": "eslint '*/**/*.{js,ts,tsx}' --fix"
+    "lint": "eslint \"*/**/*.{js,ts,tsx}\" --fix"
```
Comment (ahmed-belhadj) [inline client/package.json:34]: Why was this changed?
